documentation/src/main/asciidoc/topics/ref_indexing_configuration.adoc

```diff
@@ -0,0 +1,84 @@
+
+You can configure how {brandname} stores indexes:
+
+* On the host file system, which is the default and persists indexes between restarts.
+* In JVM heap memory, which means that indexes do not survive restarts. +
+You should store indexes in JVM heap memory only for small datasets.
+
+.File system
+[source,xml,options="nowrap",subs=attributes+]
+----
+<distributed-cache name="my-cache">
+   <indexing storage="filesystem" path="${java.io.tmpdir}/baseDir">
+     ...
+   </indexing>
+</distributed-cache>
+----
+
+.JVM heap memory
+[source,xml,options="nowrap",subs=attributes+]
+----
+<distributed-cache name="my-cache">
+  <indexing storage="local-heap">
+   ...
+</distributed-cache>
+----
+
+== Index Reader
+
+The index reader is an internal process that gets updates to the index as it changes over time.
+You can configure the refresh interval for the index reader, which controls how often {brandname} reads the index before performing a query.
+By default {brandname} reads the index before each query if the index changed since the last refresh.
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+<distributed-cache name="my-cache">
+  <indexing storage="filesystem" path="${java.io.tmpdir}/baseDir">
+    <index-reader refresh-interval="1000"/> <1>
+     ...
+</distributed-cache>
+----
+<1> Sets an interval of one second for the `IndexReader`.
+
+== Index Writer
+
+The index writer is an internal process that constructs an index that has one or more segments that can be merged over time to improve performance.
+Fewer segments usually means less overhead during a query because index reader operations need to take into account all segments.
+
+{brandname} indexes entries in two tiers: memory and storage. New entries go to the memory index first, and then when a flush happens, to the configured index storage. Periodic commit operations occur that create segments from the previously flushed data and make all the index changes permanent.
+
+[NOTE]
+====
+The `index-writer` configuration is optional.
+The defaults should work for most cases and custom configurations should only be used to tune performance.
+====
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+<distributed-cache name="my-cache">
+  <indexing storage="filesystem" path="${java.io.tmpdir}/baseDir">
+    <index-writer commit-interval="2000"
+                  low-level-trace="false"
+                  max-buffered-entries="32"
+                  queue-count="1"
+                  queue-size="10000"
+                  ram-buffer-size="400"
+                  thread-pool-size="2">
+      <index-merge calibrate-by-deletes="true"
+                   factor="3"
+                   max-entries="2000"
+                   min-size="10"
+                   max-size="20"/>
+    </index-writer>
+      ...
+</distributed-cache>
+----
+
+.Reference
+
+For more information about indexing elements and attributes, refer to the link:{configdocroot}[{brandname} Configuration Schema].
```

Comment:
Unfortunately, that xsd link is far from readable. Even pointing to the raw xsd would prompt the user to guess where to look at to find the elements, and navigate through the maze of elements, complex types, etc. 
I'd prefer to bring back the description that is in master

Reply:
sounds good. I'll add the attribute descriptions back to the doc.